neqo-transport/src/packet/mod.rs

```diff
@@ -997,12 +1012,14 @@ mod tests {
         let burn = prot.encrypt(0, &[], &[]).expect("burn OK");
         assert_eq!(burn.len(), prot.expansion());
```

Comment:
Unrelated to this PR, but these should probably return errors and not panic.

Reply:
Note that this particular line is part of a unit test. What would be the benefit of returning an error over right away panicking @larseggert?

(Agreed on your comments above on non-unit-test lines, using `debug_assert` and returning an error in release mode.)